controllers/manager/factory_install_info.go

```diff
@@ -0,0 +1,183 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/* Copyright(c) 2024 Wind River Systems, Inc. */
+
+package manager
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	perrors "github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ConfigMap to record and drive the configuration after factory install
+// Sample configMap:
+/*
+apiVersion: v1
+kind: ConfigMap
+metadata:
+	name: factory-install
+	namespace: deployment
+data:
+	factory-installed: "false"  # Initial state (change to "true" if factory-installed)
+	factory-config-finalized: "false" # Initial state (change to "true" if configuration is complete)
+	system-abcd-default-updated: "false" # Initial state (change to "true" if resource default updated)
+*/
+const FactoryInstallConfigMapName = "factory-install"
+
+func (m *PlatformManager) GetFactoryInstall(ns string) (bool, error) {
+	m.lock.Lock()
+	defer m.lock.Unlock()
+	configMap := &v1.ConfigMap{}
+	configMapName := types.NamespacedName{Namespace: ns, Name: FactoryInstallConfigMapName}
+
+	// Lookup the factory install ConfigMap from this namespace
+	err := m.GetClient().Get(context.TODO(), configMapName, configMap)
+	if err != nil {
+		if client.IgnoreNotFound(err) != nil {
+			// Any error other than NotFound is returned
+			return false, err
+		}
+		// ConfigMap not found, return false with no error
+		log.V(2).Info("Factory install configmap not found", "namespace", ns)
+		return false, nil
+	}
+
+	// Read the factory-installed status
+	factoryInstalledValue, exists := configMap.Data["factory-installed"]
```

Comment:
Will it be better to use a const for `factory-installed` key?

Reply:
Done